src/store/index.js

```diff
@@ -261,6 +206,36 @@ export default new Vuex.Store({
       } else {
         commit('setAgencies', result.agencies)
       }
+    },
+    async updateReportingPeriods ({ commit, state }) {
+      const result = await getJson('/api/reporting_periods')
+      if (result.error) {
+        commit(
+          'addAlert',
+          { text: `updateReportingPeriods Error: ${result.error} (${result.text})`, level: 'err' }
+        )
+      } else {
+        commit('setReportingPeriods', result.reportingPeriods)
+      }
+    },
+    async updateUsersRoles ({ commit, state }) {
+      const result = await getJson('/api/users')
+      if (result.error) {
+        commit('addAlert', { text: `updateUsersRoles Error: ${result.error} (${result.text})`, level: 'err' })
+      } else {
+        commit('setRoles', result.roles)
+        commit('setUsers', result.users)
+      }
+    },
+    async updateApplicationSettings ({ commit }) {
```

Comment:
I do really like this store change.  A lot easier to reason about, you're right.  Great work, thanks!